xla/service/transpose_folding.cc

```diff
@@ -101,10 +101,11 @@ absl::Status FoldTransposeIntoDot(InstructionOperandsPair& pair) {
       rhs = rhs->mutable_operand(0);
     }
   }
-
-  return dot->parent()->ReplaceWithNewInstruction(
-      dot, HloInstruction::CreateDot(dot->shape(), lhs, rhs, new_dot_dims,
-                                     dot->precision_config()));
+  HloInstruction* new_dot =
+      dot->parent()->AddInstruction(HloInstruction::CreateDot(
+          dot->shape(), lhs, rhs, new_dot_dims, dot->precision_config()));
+  new_dot->CopyBackendConfigFrom(dot);
```

Comment:
There's also metadata that can be copied. Instead of calling `CopyBackendConfigFrom`, I think calling `new_dot->SetupDerivedInstruction(dot)` is the best way to handle everything that should be copied.

Reply:
updated